src/Core/Managed/Net45/Extensibility/Implementation/RichPayloadEventSource.TelemetryHandler.cs

```diff
@@ -0,0 +1,604 @@
+                        PartB_PerformanceCounterData = new
+                        {
+                            data.ver,
+                            data.categoryName,
+                            data.counterName,
+                            data.instanceName,
+                            data.kind,
+                            data.count,
+                            data.min,
+                            data.max,
+                            data.stdDev,
+                            data.value,
+                            data.properties
+                        }
+                    };
+
+                    writeMethod.Invoke(this.EventSourceInternal, new object[] { PerformanceCounterTelemetry.TelemetryName, eventSourceOptions, extendedData });
+                }
+            };
+        }
+
+        private Action<ITelemetry> CreateHandlerForPageViewTelemetry(MethodInfo writeGenericMethod, Type eventSourceOptionsType, PropertyInfo eventSourceOptionsKeywordsProperty)
+        {
+            var eventSourceOptions = Activator.CreateInstance(eventSourceOptionsType);
+            var keywords = Keywords.PageViews;
+            eventSourceOptionsKeywordsProperty.SetValue(eventSourceOptions, keywords);
+            var dummyPageViewData = new PageViewData();
+            var writeMethod = writeGenericMethod.MakeGenericMethod(new
+            {
+                PartA_iKey = this.dummyPartAiKeyValue,
+                PartA_Tags = this.dummyPartATagsValue,
+                PartB_PageViewPerfData = new
+                {
+                    // The properties and layout should be the same as PageViewData_types.cs (EventData_types.cs)
+                    dummyPageViewData.url,
+                    dummyPageViewData.duration,
+                    dummyPageViewData.ver,
+                    dummyPageViewData.name,
+                    dummyPageViewData.properties,
+                    dummyPageViewData.measurements,
+                }
+            }.GetType());
+
+            return (item) =>
+            {
+                if (this.EventSourceInternal.IsEnabled(EventLevel.Verbose, keywords))
+                {
+                    var telemetryItem = item as PageViewTelemetry;
+                    var data = telemetryItem.Data;
+                    var extendedData = new
+                    {
+                        // The properties and layout should be the same as the anonymous type in the above MakeGenericMethod
+                        PartA_iKey = telemetryItem.Context.InstrumentationKey,
+                        PartA_Tags = telemetryItem.Context.Tags,
+                        PartB_PageViewPerfData = new
+                        {
+                            data.url,
+                            data.duration,
+                            data.ver,
+                            data.name,
+                            data.properties,
+                            data.measurements,
+                        }
```

Comment:
It will have a performance impact. If we create the method each time we emit the event, 1 million calls take 3000 ms. If we pre-create the method, it takes 300 ms.